scripts/cphd/source/copyright_update.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python
+
+#
+# =========================================================================
+# This file is part of cphd-python
+# =========================================================================
+#
+# (C) Copyright 2004 - 2019, MDA Information Systems LLC
+#
+# cphd-python is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; If not,
+# see <http://www.gnu.org/licenses/>.
+#
+
+#
+# Script to edit copyright dates and company of files
+#
+
+import sys
+import os
+import getopt
+from datetime import datetime
+
+
+# Returns list of copyright lines from file in path
+def read_copyright_txt(path):
+    data = []
+    if not os.path.isfile(path):
+        print("Failed due to invalid path: " + str(path))
+        sys.exit(2)
+    with open(path, 'r') as f:
+        data = f.readlines()
+
+    return data
+
+
+# Assumes fixed format copyright
+# Updates copyright text with year
+def update_copyright_year(crTxt, year):
+    crTxt[4] = (" * (C) Copyright 2004 - " + str(year)
+                + ", MDA Information Systems LLC\n")
+    return crTxt
+
+
+# Assumes fixed format copyright
+# Updates copyright text with module name
+def update_copyright_module(crTxt, module):
+    crTxt[1] = (" * This file is part of " + str(module) + "\n")
+    crTxt[6] = (" * " + str(module)
+                + " is free software; you can redistribute it "
+                + "and/or modify\n")
+    return crTxt
+
+
+# Fix copyright in file
+def fix_file(path, crTxt, year, module):
+    if not os.path.isfile(path):
+        print("Failed to write as path is not a file: " + str(path))
+        return
+    # Read file to be written
+    with open(path, 'r') as f:
+        data = f.readlines()
+
+    # If no copyright provided, assume copyright exists in file
+    if len(crTxt) == 0:
+        print("No Copyright text provided, assuming copyright exists in file")
+        crTxt = data[0:22]  # Assumes copyright is 22 lines (for c++)
```

Comment:
This is an assumption I'm making if someone just wants to edit an existing copyright.
Any ideas for a better way to do this? Or is this fine?

Reply:
Since this is only getting run once per year and this stuff should pretty much never change, I think it's a fine assumption to make.